src/entities/activity/ui/items/TextItem.tsx

```diff
@@ -11,10 +17,15 @@ type TextItemProps = {
 
 export const TextItem = ({ item, value, onValueChange, isDisabled }: TextItemProps) => {
   const { maxResponseLength } = item.config;
+  const [hasError, setHasError] = useState(false);
+
+  const numCharacters = value?.length || 0;
+
+  const { t } = useCustomTranslation();
 
   const onHandleValueChange = (value: string) => {
     if (value.length > maxResponseLength) {
-      return;
+      setHasError(true);
```

Comment:
`setHasError` should happen regardless of the value (similar to ParagraphText)

![CleanShot 2024-09-03 at 10 58 24@2x](https://github.com/user-attachments/assets/a86feacc-c2e2-4132-a3f6-3c482b120f67)